### PR TITLE
Fix checkpoint visiting a page that was rewritten in memory

### PR DESCRIPTION
**Pull request by michaelcahill**
refs WT-1919

### CODE REVIEW

**Comment by michaelcahill** [comment]
@keithbostic, can you please review this one?

File: src/evict/evict_page.c [hunk 1]
```diff
@@ -195,25 +195,10 @@ __evict_page_dirty_update(WT_SESSION_IMPL *session, WT_REF *ref, int exclusive)
...
-		 * Second, a real split where we reconciled a page and it turned
-		 * into a lot of pages.
+		 * A real split where we reconciled a page and it turned into a
+		 * lot of pages.
 		 */
-		if (mod->mod_multi_entries == 1)
-			WT_RET(__wt_split_rewrite(session, ref));
-		else
-			WT_RET(__wt_split_multi(session, ref, exclusive));
+		WT_RET(__wt_split_multi(session, ref, exclusive));
```
Comment (agorrod) [inline src/evict/evict_page.c]: Is it OK to call __wt_split_multi if there happens to only be a single page? Is it ever possible for ```WT_PM_REC_MULTIBLOCK``` to be set, even though the split only generated a single page?

Comment (michaelcahill) [inline src/evict/evict_page.c]: I don't believe we will ever get here (in the branch) with a single page and `WT_PM_REC_MULTIBLOCK` set.  We check how many pages were created during reconciliation, and set `EMPTY` for zero, `REPLACE / REWRITE` for one and `MULTIBLOCK` for more than one.

Even if we did somehow end up here, we would have come via `__rec_split_col/row` so everything would be set up as expected by `__rec_col/row_merge`.

**Comment by agorrod** [comment]
Thanks for the explanation. It'd be nice to get this one into stress testing. I'll merge.

@keithbostic It would be good if you can look through this change as well.

**Comment by keithbostic** [comment]
@michaelcahill, this comment:
```
       /*
         * Allow for the splitting of pages when a checkpoint is underway only
         * if the allow_splits flag has been passed, we know we are performing
         * a checkpoint, the page is larger than the stated maximum and there
         * has not already been a split on this page as the WT_PM_REC_MULTIBLOCK
         * flag is unset.
         */
```
in `__wt_page_can_evict` no longer matches the code (or the called function).

**Comment by keithbostic** [comment]
lgtm.

**Pull request merged**